### DISCUSSION TITLE
Questions regarding ENAS algo

### DISCUSSION THREAD

**Discussion by JiaojiaoYe1994** [microsoft/nni#3757] [Q&A] [answered]
I have two questions regrading implementation of ENAS:

1. How is the parameter sharing of child models implemented in ENAS of NNI?
2. Why in micro search mode, there is two same inputs for one node, i.e cell_x and cell_y in `example/nas/enas/micro.py`.

```
class Node(mutables.MutableScope):
    def __init__(self, node_name, prev_node_names, channels):
        super().__init__(node_name)
        self.cell_x = Cell(node_name + "_x", prev_node_names, channels)
        self.cell_y = Cell(node_name + "_y", prev_node_names, channels)
```

**Comment by ultmaster** [comment, marked as answer]
1. Parameter sharing can be implemented by nature in PyTorch. Use the same convolution and forward-prop with different connections in `forward`.
2. Please read ENAS paper. In each node there are two operators and two inputs. The cell here is not the **cell** that is usually used in literature.

**Comment by JiaojiaoYe1994** [comment]
Got it, thank you for the answer!  Will distributed training be implemented?

**Comment by ultmaster** [comment]
Do you mean distributed search or distributed retrain?

**Comment by zhanghang1989** [comment]
> Do you mean distributed search or distributed retrain?

Does NNI support distributed search?
